applications/Healthcare_chatbot/main.py

```diff
@@ -0,0 +1,96 @@
+import os
+import uvicorn
+from fastapi import FastAPI
+from pydantic import BaseModel
+from langchain import hub
+import logging
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from langchain_community.llms import LlamaCpp
+from langchain_community.vectorstores import LanceDB
+from langchain.document_loaders import PyPDFLoader
+from langchain_community.embeddings import SentenceTransformerEmbeddings
+from langchain_openai import ChatOpenAI
+from langchain.chains import RetrievalQA
+from langchain_cohere import CohereRerank
+from langchain.retrievers.contextual_compression import ContextualCompressionRetriever
+from langchain_cohere import CohereRerank
+from fastapi.middleware.cors import CORSMiddleware
+from langchain_community.document_loaders import PyPDFLoader, DirectoryLoader
+from langchain_community.llms import LlamaCpp
+
+
+# Environment variables for sensitive information
+OPENAI_API_KEY = os.getenv("OPENAI_API_KEY", "")
+COHERE_API_KEY = os.getenv("COHERE_API_KEY", "")
+
+app = FastAPI(title="Medical Information Retrieval API")
+
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],  
+    allow_credentials=True,
+    allow_methods=["*"],  
+    allow_headers=["*"],  
+)
+
+
+# Load the document
+DATA_PATH = "data/"
+
+
+loader = DirectoryLoader(DATA_PATH,
+                             glob='*.pdf',
+                             loader_cls=PyPDFLoader)
+
+docs = loader.load()
+logging.info("Document loader done.")
+
+# Set up the text processing and model chain
+#llm = ChatOpenAI(model="gpt-4", temperature=0, openai_api_key=OPENAI_API_KEY)
+
+# download weights from https://huggingface.co/PrunaAI/OpenBioLLM-Llama3-8B-GGUF-smashed/tree/main
+llm = LlamaCpp(
+    model_path="/content/OpenBioLLM-Llama3-8B.Q4_K_S.gguf",  # path of gguf weight
```

Comment:
How this .gguf file has to be downloaded?

Reply:
added it in readme